probe-rs/src/architecture/arm/memory/adi_v5_memory_interface.rs

```diff
@@ -370,16 +393,38 @@ where
         Ok(result.data)
     }
 
+    /// Read an 16 bit word at `address`.
+    ///
+    /// The address where the read should be performed at has to be word aligned.
+    /// Returns `ArmError::MemoryNotAligned` if this does not hold true.
+    pub fn read_word_16(&mut self, access_port: MemoryAp, address: u64) -> Result<u16, ArmError> {
+        if self.ap_information.supports_only_32bit_data_size {
```

Comment:
I think this is a bit limiting, I think support for 16 bit access and 8 bit accesses is independent. So we should probably change this flag.

(Maybe out of scope for this PR)

Reply:
according to ADIv5, MEM-APs support either only 32bit, or 32bit+16bit+8bit, so I think this is correct?

Reply:
That's correct. It's either all or none :)

Reply:
marking as resolved then